exe-unit/examples/transfer_resume.rs

```diff
@@ -0,0 +1,166 @@
+    fn schemes(&self) -> Vec<&'static str> {
+        self.inner.schemes()
+    }
+
+    fn source(&self, url: &Url, ctx: &TransferContext) -> TransferStream<TransferData, Error> {
+        let mut src = self.inner.source(url, ctx);
+        let interval = self.interval;
+        let failure = self.last_failure.clone();
+
+        src.map_inner(move |r| match r {
+            Ok(v) => {
+                let instant = { *failure.lock().unwrap() };
+                if Instant::now() - instant >= interval {
+                    log::info!("Triggering failure");
+
+                    let mut guard = failure.lock().unwrap();
+                    *guard = Instant::now();
+
+                    Err(HttpError::Io(ErrorKind::Interrupted).into())
+                } else {
+                    Ok(v)
+                }
+            }
+            Err(e) => Err(e),
+        });
+
+        src
+    }
+
+    fn destination(&self, url: &Url, ctx: &TransferContext) -> TransferSink<TransferData, Error> {
+        self.inner.destination(url, ctx)
+    }
+
+    fn prepare_source<'a>(
+        &self,
+        url: &Url,
+        ctx: &TransferContext,
+    ) -> LocalBoxFuture<'a, Result<(), Error>> {
+        self.inner.prepare_source(url, ctx)
+    }
+
+    fn prepare_destination<'a>(
+        &self,
+        url: &Url,
+        ctx: &TransferContext,
+    ) -> LocalBoxFuture<'a, Result<(), Error>> {
+        self.inner.prepare_destination(url, ctx)
+    }
+}
+
+fn hash_file<P: AsRef<Path>>(path: P) -> GenericArray<u8, <sha3::Sha3_224 as Digest>::OutputSize> {
+    const CHUNK_SIZE: usize = 40960;
+
+    let mut file_src = OpenOptions::new().read(true).open(path).expect("rnd file");
+    let mut hasher = sha3::Sha3_224::default();
+    let mut chunk = vec![0; CHUNK_SIZE];
+
+    while let Ok(count) = file_src.read(&mut chunk[..]) {
+        hasher.input(&chunk[..count]);
+        if count != CHUNK_SIZE {
+            break;
+        }
+    }
+    hasher.result()
+}
+
+async fn download<P: AsRef<Path>>(dst_path: P, args: Cli) -> anyhow::Result<()> {
+    let dst_path = dst_path.as_ref();
+
+    let src_url = TransferUrl::parse(&args.url, "http")?;
+    let dst_url = TransferUrl::parse(&dst_path.display().to_string(), "file")?;
+    let src = Rc::new(UnreliableHttpProvider::new(args.interval));
+    let dst = Rc::new(FileTransferProvider::default());
+    let ctx = TransferContext::default();
+
+    let mut retry = Retry::new(i32::MAX);
+    retry.backoff(1., 1.);
+    ctx.state.retry_with(retry);
+
+    transfer_with(&src, &src_url, &dst, &dst_url, &ctx).await?;
+
+    log::info!("File downloaded, verifying contents");
+
+    let hash = hex::encode(hash_file(&dst_path));
+
+    log::info!("input  hash: {}", args.hash);
+    log::info!("result hash: {}", hash);
+
+    if args.hash != hash {
+        anyhow::bail!("hash mismatch");
+    }
+    Ok(())
+}
+
+#[actix_rt::main]
+async fn main() -> anyhow::Result<()> {
```

Comment:
When input url points to not existing machine, it is resulting with:
```
http://yacn3.dev.golem.network:8000/docker-yalife-0.3-fa5b098935.gvmi' b3fd75d65e822703f95e717034e9db84b9a1487aee915a0d660b7eb2`
[2021-06-17T13:43:18Z DEBUG ya_transfer] Transferring from offset: 0
[2021-06-17T13:43:18Z DEBUG ya_transfer::file] Transferring to file: /var/folders/8v/ssjr4gw17hn1fpkq4rjwmmxw0000gn/T/transfer-resume.GCLAyZ5cuDoB/downloaded
[2021-06-17T13:43:19Z WARN  ya_transfer] Retrying in 1s: HTTP error: timeout: connection
```

But should:
```
$ curl http://yacn3.dev.golem.network:8000/docker-yalife-0.3-fa5b098935.gvmi 
curl: (6) Could not resolve host: yacn3.dev.golem.network
```

Reply:
`trust_dns_resolver` times out while resolving that name. No distinguishable error is returned in this case for your OS.

For example, on Ubuntu 21.04 I get:
```
HTTP error: connection error: Failed resolving hostname: no record found for name: yacn3.dev.golem.network.home. type: AAAA class: IN
```

For the sake of retrying transfers, I'd keep this mechanism as is and revisit after upgrading `awc` to `3.x`

Reply:
yeah, ok